dev-lang/rust/rust-1.74.1-r100.ebuild

```
@@ -504,7 +530,257 @@ src_configure() {
+	mkdir -p "${stage2_sysroot_lib}" || die "Failed to create stage2 directory"
+	mkdir -p "${stage2}/bin" || die
+
+	# Copy required files from stage1 to stage2 sysroot
+	cp "${stage1}/rustc-build/${CFG_COMPILER_HOST_TRIPLE}"/release/rustc-main "${stage2}/bin/rustc_binary" ||
+		die "Failed to copy final rustc to stage2 sysroot"
+	cp "${stage1}/rustc-build/${CFG_COMPILER_HOST_TRIPLE}"/release/librustc_driver.so "${stage2}/lib" ||
+		die "Failed to copy librustc_driver to sysroot"
+	cp "${stage1}/rustc-build/${CFG_COMPILER_HOST_TRIPLE}"/release/deps/*.{rlib,so} "${stage2_sysroot_lib}" ||
+		die "Failed to copy final rustc libs to stage2 sysroot"
+	cp "${stage1_sysroot_lib}"/* "${stage2_sysroot_lib}" || die "Failed to copy stage1 so files to stage2 sysroot"
+
+	# There's a magic script used in place of rustc so that libs can be found
+	cat << EOF > "${stage2}/bin/rustc"
+#!/bin/sh
+LD_LIBRARY_PATH="${stage2}/lib:${stage2_sysroot_lib}" ${stage2}/bin/rustc_binary "\$@"
+EOF
+	chmod +x "${stage2}/bin/rustc" || die "Failed to make rustc wrapper executable"
+
+	# Use rustc to build sysroot; this is the final step in the bootstrap process.
+	# rpath probably isn't needed here, but it doesn't hurt.
+	RUSTFLAGS="-Z force-unstable-if-unmarked -C link_args=-Wl,-rpath,\$ORIGIN/../lib" \
+		CARGO_TARGET_DIR="${stage2}/stdlib-build" RUSTC="${stage2}/bin/rustc" \
+		"${stage0}"/cargo-build/cargo build ${cargo_flags} --manifest-path "${S}/library/sysroot/Cargo.toml" \
+		--features panic-unwind ||
+			die "Failed to build sysroot (mrustc-stage2)"
+
+	# Build a coherent output dir.
+	local output="${WORKDIR}/bootstrap/rust-${PV}"
+	local output_sysroot_lib="${output}/lib/rustlib/${CFG_COMPILER_HOST_TRIPLE}/lib"
+	mkdir -p "${output_sysroot_lib}" || die "Failed to create output directory"
+	mkdir -p "${output}/bin" || die "Failed to create output directory"
+
+	# Copy our various output files into the final output directory
+	# rustc
+	cp "${stage1}/rustc-build/${CFG_COMPILER_HOST_TRIPLE}"/release/rustc-main "${output}/bin/rustc_binary" ||
+		die "Failed to copy final rustc to output"
+	cp "${stage1}/rustc-build/${CFG_COMPILER_HOST_TRIPLE}"/release/librustc_driver.so "${output}/lib" ||
+		die "Failed to copy librustc_driver to output"
+	cp "${stage1}/rustc-build/${CFG_COMPILER_HOST_TRIPLE}"/release/deps/*.{rlib,so} "${output_sysroot_lib}" ||
+		die "Failed to copy final rustc libs to output"
+	# cargo; no need to build an optimised cargo if we're using this to build a complelety new Rust.
+	cp "${stage0}/cargo-build/cargo" "${output}/bin/cargo" || die "Failed to copy cargo to output"
+	# sysroot
+	mv "${stage2}/stdlib-build/${CFG_COMPILER_HOST_TRIPLE}/release/deps"/*.{rlib,rmeta,so} "${output_sysroot_lib}" ||
+		die "Failed to copy stage2 libs to output"
+	# Our trusty rustc wrapper
+	cat << EOF > "${output}/bin/rustc"
+#!/bin/sh
+LD_LIBRARY_PATH="${output}/lib:${output_sysroot_lib}" ${output}/bin/rustc_binary "\$@"
+EOF
```

I think this makes it a little easier to read:
```suggestion
	cat <<- EOF > "${output}/bin/rustc" || die
		#!/bin/sh
		LD_LIBRARY_PATH="${output}/lib:${output_sysroot_lib}" ${output}/bin/rustc_binary "\$@"
	EOF
```